File: app/src/main/java/com/typingsolutions/kore/setup/SimpleSetupFragment.java
package com.typingsolutions.kore.setup;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.TextInputEditText;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import com.typingsolutions.kore.R;

public class SimpleSetupFragment extends Fragment implements IPasswordProvider {
  private TextInputEditText mEditTextAsEnterPassword;
  private TextInputEditText mEditTextAsRepeatPassword;

  @Nullable
  @Override
  public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
    View root = inflater.inflate(R.layout.setup_fragment_2, container, false);
    mEditTextAsEnterPassword = (TextInputEditText) root.findViewById(R.id.setuplayout_edittext_passwordenter);
    mEditTextAsRepeatPassword = (TextInputEditText) root.findViewById(R.id.setuplayout_edittext_passwordrepeat);

    mEditTextAsEnterPassword.addTextChangedListener(new EnableSetupTextWatcher((SetupActivity)getActivity(), mEditTextAsRepeatPassword));
    mEditTextAsRepeatPassword.addTextChangedListener(new EnableSetupTextWatcher((SetupActivity)getActivity(), mEditTextAsEnterPassword));

    return root;
  }

  @Override
  public CharSequence getPassword1() {
    return mEditTextAsEnterPassword.getText();
  }

  @Override
  public CharSequence getPassword2() {
    return mEditTextAsRepeatPassword.getText();
  }

  @Override
  public void setPasswords(CharSequence pw1, CharSequence pw2) {

  }

  @Override
  public void cleanUp() {
    SetupActivity activity = (SetupActivity) getActivity();

    activity.clearText(mEditTextAsEnterPassword);
    activity.clearText(mEditTextAsRepeatPassword);

    mEditTextAsEnterPassword.clearComposingText();
    mEditTextAsRepeatPassword.clearComposingText();

    mEditTextAsEnterPassword = null;
    mEditTextAsRepeatPassword = null;
  }
}

File: app/src/main/java/com/typingsolutions/kore/setup/SimpleSetupFragment.kt
package com.typingsolutions.kore.setup

import android.os.Bundle
import android.support.design.widget.TextInputEditText
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.typingsolutions.kore.R
import com.typingsolutions.kore.common.ViewUtil

class SimpleSetupFragment : Fragment(), IPasswordProvider {
    private lateinit var mEditTextAsEnterPassword: TextInputEditText
    private lateinit var mEditTextAsRepeatPassword: TextInputEditText

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val root = inflater!!.inflate(R.layout.setup_fragment_2, container, false)
        mEditTextAsEnterPassword = root.findViewById(R.id.setuplayout_edittext_passwordenter) as TextInputEditText
        mEditTextAsRepeatPassword = root.findViewById(R.id.setuplayout_edittext_passwordrepeat) as TextInputEditText

        mEditTextAsEnterPassword.addTextChangedListener(EnableSetupTextWatcher(activity as SetupActivity, mEditTextAsRepeatPassword))
        mEditTextAsRepeatPassword.addTextChangedListener(EnableSetupTextWatcher(activity as SetupActivity, mEditTextAsEnterPassword))

        return root
    }

    override fun getPassword1(): CharSequence? {
        return mEditTextAsEnterPassword.text
    }

    override fun getPassword2(): CharSequence? {
        return mEditTextAsRepeatPassword.text
    }

    override fun setPasswords(pw1: CharSequence, pw2: CharSequence) {

    }

    override fun cleanUp() {
        ViewUtil.clearText(mEditTextAsEnterPassword)
        ViewUtil.clearText(mEditTextAsRepeatPassword)

        mEditTextAsEnterPassword.clearComposingText()
        mEditTextAsRepeatPassword.clearComposingText()
    }
}
